arch/x86/core/ia32/crt0.S

```diff
@@ -235,6 +235,10 @@ __csSet:
 #endif
 	lidt	z_x86_idt		/* load 32-bit operand size IDT */
 
+#ifdef CONFIG_LOAPIC
+	call	z_loapic_enable
```

Comment:
I'm wondering if we can just put this call in z_x86_prep_c instead of here, since it appears that for both 32 and 64 we just call z_x86_prep_c immediately afterwards anyway.

Reply:
On 64-bit systems, only the BSP lands in the kernel at `z_x86_prep_c()` .. the rest of the processors enter at `smp_init_top()`.  Short story is that we could do what you suggest, and consolidate the code paths in this way, but that means splitting them somewhere else.  It's sorta ugly either way, unfortunately.

Reply:
Ah OK, let's leave it.